src/compas_ags/ags/graphstatics.py

```diff
@@ -416,4 +415,38 @@ def force_update_from_form(force, form):
 # ==============================================================================
 
 if __name__ == "__main__":
-    pass
+    import compas_ags
+    from compas_ags.diagrams import FormGraph
+    from compas_ags.diagrams import FormDiagram
+    from compas_ags.diagrams import ForceDiagram
+    from compas_ags.viewers import Viewer
+
+
+    graph = FormGraph.from_obj(compas_ags.get('paper/gs_form_force.obj'))
+    form = FormDiagram.from_graph(graph)
+    force = ForceDiagram.from_formdiagram(form) 
+    form.edge_force((0, 1), -1.0)
+
+    form.data = form_update_q_from_qind_proxy(form.data)
+    force.data = force_update_from_form_proxy(force.data, form.data)
+
+    print(form_count_dof_proxy(form.data))
+    print(form_identify_dof_proxy(form.data))
+
+    viewer = Viewer(form, force, delay_setup=False, figsize=(8, 5))
+
+    left  = list(form.vertices_where({'x': 0.0, 'y': 0.0}))[0]
+    right = list(form.vertices_where({'x': 6.0, 'y': 0.0}))[0]
+
+    viewer.draw_form(
+        vertexsize=0.15,
+        vertexcolor={key: '#000000' for key in (left, right)},
+        vertexlabel={key: key for key in form.vertices()},
+        edgelabel={uv: index for index, uv in enumerate(form.edges())})
+
+    viewer.draw_force(
+        vertexsize=0.15,
+        vertexlabel={key: key for key in force.vertices()})
+
+    viewer.show()
```

Comment:
using the main section for examples is deprecated (since a while, but i may not have been corrected everywhere yet). if anything, please use for running `doctest` snippets, and add the examples in an example or test file...